qiskit/providers/ibmq/api_v2/rest/schemas/auth.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2018.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Model and schema for authentication."""
+from marshmallow.fields import String, Url, Dict
+from qiskit.validation import BaseSchema, bind_schema, BaseModel
+
+
+class TokenSchema(BaseSchema):
+    """Schema for Token."""
+
+    # Required properties.
+    token_id = String('id', required=True)
+
+
+@bind_schema(TokenSchema)
+class Token(BaseModel):
+    """Model for Token.
+
+    Please note that this class only describes the required fields. For the
+    full description of the model, please check ``TokenSchema``.
+
+    Attributes:
+        token_id (str): long term access token.
+    """
+
+    def __init__(self, token_id, **kwargs):
+        self.token_id = token_id
+
+        super().__init__(**kwargs)
+
+
+class UserSchema(BaseSchema):
+    """Schema for User."""
+
+    # Required properties.
+    urls = Dict(Url(), required=True)
```

Comment:
Can you detail more this `urls` field by changing it into a Schema of its own that becomes `Nested`? This is one of the things we want to achieve via the documentation effort - currently, the provider expects the data that comes from this method contains at least two keys with specific names.

Reply:
There is a problem with the application of `Nested`.

Initially, I was trying to express this schema this way:
```
class UserApiUrlResponseSchema(BaseSchema):
    """Nested schema for UserInfoResponse"""

    # Required properties.
    protocol = fields.String(required=True, metadata='api protocol')
    url = fields.Url(required=True, metadata='URL address to the api')


class UserInfoResponseSchema(BaseSchema):
    """Schema for UserInfoResponse."""

    # Required properties.
    urls = fields.Nested(UserApiUrlResponseSchema, many=True, required=True,
                         metadata='urls and corresponding protocols associated with APIs')
```

But then I realized that those URLs are stored as a map:
```
{
    ...,
    "urls": {
            "http": "https://api.quantum-computing.ibm.com/api", 
            "ws": "wss://wss.quantum-computing.ibm.com/"
    },
    ...
}
```

In order for my schema to be deserialized correctly, it should have been looked like this:
```
{
    ...,
    "urls": {
            [
                { "protocol": "http", "url": "https://api.quantum-computing.ibm.com/api" },
                { "protocol": "ws", "url": "wss://wss.quantum-computing.ibm.com/"}
            ]
    },
    ...
}
```

That's why nested schema approach might not serve for our purposes.

Thinking about possible solutions...

Reply:
Found similar issue raised under marshmallow project:
https://github.com/marshmallow-code/marshmallow/issues/483

Reply:
Ok, I think I found a right combination:
```
class UserApiUrlResponseSchema(BaseSchema):
    """Nested schema for UserInfoResponse"""

    # Required properties.
    protocol = fields.String(required=True, metadata='api protocol')
    url = fields.Url(required=True, metadata='URL address to the api')


class UserInfoResponseSchema(BaseSchema):
    """Schema for UserInfoResponse."""

    # Required properties.
    urls = fields.Dict(fields.Nested(UserApiUrlResponseSchema), many=True, required=True,
                       metadata='urls and corresponding protocols associated with APIs')
```